receiver/otlpreceiver/internal/arrow/arrow.go

```diff
@@ -141,15 +141,28 @@ func (h *headerReceiver) combineHeaders(ctx context.Context, hdrsBytes []byte) (
 
 	h.tmpHdrs = map[string][]string{}
 
-	for k, v := range h.streamHdrs {
-		h.tmpHdrs[k] = v
-	}
-
 	// Write calls the emitFunc, appending directly into `tmpHrs`.
 	if _, err := h.decoder.Write(hdrsBytes); err != nil {
 		return ctx, err
 	}
 
+	// Add streamHdrs that were not carried in the per-request headers.
```

Comment:
I don't know if it's important in this context, but it's technically possible (and sometimes useful) to have several HTTP headers with the same key but with different values. This code deduplicates the headers only on the keys, I wonder if we shouldn't also deduplicate on the values. Not knowing exactly the problem we are trying to solve, my remark is maybe not relevant.

Reply:
This is a fine point and there are loose ends described in https://github.com/open-telemetry/opentelemetry-collector/issues/6965. The exporter presently calls the `Auth` module twice, each time calling through to a gRPC auth-related feature that returns a `map[string]string` (single value). Meanwhile, the HTTP headers are `map[string][]string` (list of values).

The overwrite happening in this PR is replacing the keys of one `map[string][]string` for another, the result being that multiple key-values are supported. However, the source of this information doesn't support returning multiple values and indeed the `client.Metadata` type used in the collector will return an error if you try to `Get()` and there are more than one value. The reason we have multiple values is because we're calling the Auth extension twice. Without this duplication we would either have two copies of the same value, or we would have one empty string and one correct value. This is a workaround -- if the `Headers` configuration field of the exporter listed multiple values for a header (if that's allowed), they would pass through as multi-valued correctly.

I would call this an unsolved problem. In the issue linked above I've hinted at a solution I like. We should separate the auth-related production of metadata from ordinary metadata. Ordinary metadata can be `[]string` and auth-related metadata cannot, both are established conventions. It suggests might want to add a new extension type for producing outgoing metadata not from an Auth extension. If this were done, the Arrow stream would call only the second of these, and there would be no accidental duplication.